### PR TITLE
level aa contrast home page

### PR DESCRIPTION
**Pull request by ATBull81**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [x] Tests for the changes have been added (for bugfixes / features)

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update version)

# What is the ticket # detailing the issue?

Ticket: https://www.pivotaltracker.com/story/show/186499314 && https://www.pivotaltracker.com/story/show/186499197

# A brief description of the changes

Current behavior No cucumber, a few contrast issues, and missing tooltip fallback and hover

New behavior: All behaviors improved and cucumber test added

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable that is used to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name: CONTRAST_LEVEL_AA_IS_ENABLED

- [ ] DC
- [x] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

Screenshot 2024-01-10 at 2 31 29 PM
Screenshot 2024-01-10 at 2 32 11 PM
Screenshot 2024-01-10 at 2 32 16 PM

This PR also contains a fix for a glossary issue in the FAA workflow. The new functionality is tested and shown to work locally:
Screenshot 2024-01-10 at 3 42 30 PM
Screenshot 2024-01-10 at 3 43 11 PM
Screenshot 2024-01-10 at 6 34 10 PM

On that same page, some radio buttons were missed:
Screenshot 2024-01-10 at 8 16 43 PM

# AppScan CodeSweep Failure
In the event of a failed check on the AppScan CodeSweep step of our GitHub Actions workflow, please review the False Positive protocol outlined here: appscan_codesweep/CODESWEEP_FALSE_POSITIVES_README.MD

Add all required notes to this section if the failure is a suspected false positive.

### CODE REVIEW

File: app/views/layouts/two_column.html.erb [hunk 1]
```diff
@@ -14,6 +14,11 @@
     <%= javascript_pack_tag 'legacy', 'data-turbolinks-track': 'reload' %>
     <%= csrf_meta_tags %>
 
+    <% if EnrollRegistry.feature_enabled?(:contrast_level_aa) %>
+      <%= javascript_pack_tag 'contrast_level_aa' %>
+      <%= stylesheet_pack_tag 'contrast_level_aa' %>
+    <% end %>
+
```
Comment (ATBull81) [inline app/views/layouts/two_column.html.erb:21]: Once you hit the homepage, this was required for the new layout

File: components/financial_assistance/app/views/financial_assistance/benefits/index.html.erb [hunk 2]
```diff
@@ -34,7 +34,7 @@
     </div>
 
     <div class='benefit-kinds', id='enrolled-benefit-kinds'>
-      <%= render partial: 'financial_assistance/benefits/insurance_kinds', locals: { insurance_kinds: FinancialAssistance::Benefit.valid_insurance_kinds, kind: "is_enrolled", text: "What type of coverage is this person currently enrolled in? *"} %>
+      <%= render partial: 'financial_assistance/benefits/insurance_kinds', locals: { insurance_kinds: FinancialAssistance::Benefit.valid_insurance_kinds, kind: "is_enrolled", text1: "What type of ", text2: " is this person currently enrolled in? *"} %>
```
Comment (ATBull81) [inline components/financial_assistance/app/views/financial_assistance/benefits/index.html.erb:37]: The way the rendering was being handled in _insurance_kinds.html.erb, you couldn't use the tabindex on the glossary word. This was the most straightforward fix.

File: features/hbx_admin/step_definitions/ui_config_steps.rb [hunk 3]
```diff
@@ -159,3 +159,11 @@
...
   sleep(1)
 end
+
+Then(/^the page is axe clean$/) do
+  # Ensures the axecore contrast tests are not run immediately after the DOM is available but prior to the page
+  # fully rendering in the browser.
+  # USE THIS ONLY AFTER CONFIRMING THAT IT IS ABSOLUTELY NECESSARY
+  sleep(1)
+  steps %(the page should be axe clean excluding "a[disabled], .disabled" according to: wcag2aa; checking only: color-contrast)
+end
```
Comment (ATBull81) [inline features/hbx_admin/step_definitions/ui_config_steps.rb:169]: We have seen builds fail randomly along with a bug in trunk previously caused by a build passing with a false positive. Moving forward, we will be including the sleep in all axe clean checks using this step.

File: components/financial_assistance/app/views/financial_assistance/benefits/_insurance_kinds.html.erb [hunk 4]
```diff
@@ -43,7 +43,13 @@
...
     </div>
     <div class="col-md-12 add-more-link-<%=insurance_kind%>" id="add-more-link">
-      <%= link_to "<span class='fa-icon fa-stack plus-mr' aria-hidden='true'><i class= 'fa fa-circle-thin fa-stack-2x'></i> <i class='fa fa-plus fa-stack-1x'></i></span> <b>Add More</b> #{l10n("faa.question.#{insurance_kind}", short_name: EnrollRegistry[:enroll_app].setting(:short_name).item)}".html_safe, "javascript:void(0);", :id => "add_new_insurance_kind", :class => "#{'hidden' unless @applicant.benefits.where(kind: kind).of_insurance_kind(insurance_kind).present? }" %>
+      <a tabindex="0" onkeydown="handleButtonKeyDown(event, 'add_new_insurance_kind')" id="add_new_insurance_kind" class="<%= 'hidden' unless @applicant.benefits.where(kind: kind).of_insurance_kind(insurance_kind).present? %>">
+        <span class="fa-icon fa-stack plus-mr" aria-hidden="true">
+          <i class="fa fa-circle-thin fa-stack-2x"></i>
+          <i class="fa fa-plus fa-stack-1x"></i>
+        </span>
+        <b> <%= l10n("add_more") %> </b> <%= l10n("faa.question.#{insurance_kind}", short_name: EnrollRegistry[:enroll_app].setting(:short_name).item) %>
+      </a>
```
Comment (ATBull81) [inline components/financial_assistance/app/views/financial_assistance/benefits/_insurance_kinds.html.erb:52]: improved the add more button logic in several places to make it more readable and pass codesweep

**Review by scaustin34 — APPROVED** [review]
Looks good!

**Review by mdkaraman — APPROVED** [review]
Nice work on the refactors!  And very thorough with the attached comments 👏

**Pull request merged**